files/image_config/caclmgrd/caclmgrd

```diff
@@ -347,13 +362,21 @@ class ControlPlaneAclManager(object):
                         # try to do it now. We attempt to determine heuristically based on
                         # whether the src or dst IP of this rule is an IPv4 or IPv6 address.
                         if not table_ip_version:
-                            if (("SRC_IPV6" in rule_props and rule_props["SRC_IPV6"]) or
-                                ("DST_IPV6" in rule_props and rule_props["DST_IPV6"])):
+                            if self.is_rule_ipv6(rule_props):
                                 table_ip_version = 6
-                            elif (("SRC_IP" in rule_props and rule_props["SRC_IP"]) or
-                                  ("DST_IP" in rule_props and rule_props["DST_IP"])):
+                            elif self.is_rule_ipv4(rule_props):
                                 table_ip_version = 4
 
+                        if (self.is_rule_ipv6(rule_props) and (table_ip_version == 4)):
+                            log_error("CtrlPlane ACL table {} is a IPv4 based table and rule {} is a IPV6 rule!"
+                                                          .format(table_name, rule_id))
+                            raise CPACLTblRulesException("Trying to insall IPv6 rule %s in a IPv4 based ACL table"%rule_id)
```

Comment:
Should we throw an exception here, or should we simply log a warning message and skip the rule?

Reply:
It was a comment from Stepan to raise an exception and not just logging a warning.

Reply:
@stepanblyschak: What benefit do you see in throwing an exception here? I feel that we should probably simply log a warning and continue applying other ACL rules in order to secure the device as much as possible, rather than terminating the daemon altogether and potentially leaving the device unsecured.

Reply:
My comment (during internal review) was about throwing exception vs sys.exit(1). Your comment is reasonable, so I agree about logg an ERROR.